File: Data Structures & Algorithms/daily-temperatures/submission-1.js
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures) {
        const stack = []
        const result = new Array(temperatures.length).fill(0)
        for(let r=temperatures.length-1;r>=0;r--){
            while(stack && temperatures[stack[stack.length-1]]<=temperatures[r]){
                stack.pop()
            }
            let distance = (stack[stack.length-1] - r) || 0
            result[r]=(distance)
            stack.push(r)
        }
        return result
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-1.py
class Solution:
    def isValid(self, s: str) -> bool:
        close = {
            ')' : '(',
            '}' : '{',
            ']' : '['
        }
        stack = []
        for let in s:
            if let in close:
                if stack and stack[-1]==close[let]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(let)
        return True if not stack else False